File: src/main/java/my/study/repository/design/pattern/strategy/java/Context.java
package my.study.repository.design.pattern.strategy.java;

/**
 * @author djh on  2018/9/12 16:09
 * @E-Mail 1544579459@qq.com
 */
public class Context {

    int calculate(Strategy strategy, int distance) {
        return strategy.calculatePrice(distance);
    }

}

File: src/main/java/my/study/repository/design/pattern/strategy/kotlin/Context.kt
package my.study.repository.design.pattern.strategy.kotlin

/**
 * @author djh on  2018/10/15 19:36
 * @E-Mail 1544579459@qq.com
 */
class Context {
    fun calculate(strategy: (distance: Int) -> Float, distance: Int): Float = strategy(distance)
}